Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
```

Comment:
I think the way forward here, which will also massively simplify the logic and avoid having separate methods for Python 2 and Python 3 is to just use `Requirement#name`, i.e. modify this check to be `req.name == "python"` or `req.name == "python3"`.